mage_ai/io/pinot.py

```diff
@@ -0,0 +1,163 @@
+            port (int): Port on which the pinot controller is running.
+            username (str): The user with which to connect to the server with.
+            password (str): The login password for the user.
+            path (str): Path to pinot sql api.
+            scheme (str): The scheme identifies the protocol to be used http or https.
+            **kwargs: Additional settings for creating SQLAlchemy engine and connection
+        """
+        self._ctx = None
+        super().__init__(
+            host=host,
+            port=port,
+            username=username,
+            password=password,
+            path=path,
+            scheme=scheme,
+            **kwargs,
+        )
+
+    @classmethod
+    def with_config(cls, config: BaseConfigLoader) -> 'Pinot':
+        return cls(
+            host=config[ConfigKey.PINOT_HOST],
+            port=config[ConfigKey.PINOT_PORT],
+            username=config[ConfigKey.PINOT_USER],
+            password=config[ConfigKey.PINOT_PASSWORD],
+            path=config[ConfigKey.PINOT_PATH],
+            scheme=config[ConfigKey.PINOT_SCHEME],
+        )
+
+    def open(self) -> None:
+        with self.printer.print_msg('Opening connection to Pinot warehouse'):
+            connect_kwargs = dict(
+                host=self.settings['host'],
+                port=self.settings['port'],
+                username=self.settings['username'],
+                password=self.settings['password'],
+                path=self.settings['path'],
+                scheme=self.settings['scheme'],
+            )
+            self._ctx = ConnectionWrapper(**connect_kwargs)
+
+    def execute(self, query_string: str, **query_vars) -> None:
+        super().execute(query_string, **query_vars)
+
+    def execute_queries(self,
+                        queries: List[str],
+                        query_variables: List[Dict] = None,
+                        commit: bool = False,
+                        fetch_query_at_indexes: List[bool] = None) -> List:
+        return super().execute_queries(queries, query_variables, commit, fetch_query_at_indexes)
+
+    def fetch_query(self, cursor, query: str) -> Any:
+        return super().fetch_query(cursor, query)
+
+    def load(self,
+             query_string: str,
+             limit: int = QUERY_ROW_LIMIT,
+             display_query: Union[str, None] = None,
+             verbose: bool = True,
+             **kwargs) -> DataFrame:
+        return super().load(query_string, limit, display_query, verbose, **kwargs)
+
+    def clean(self, column: Series, dtype: str) -> Series:
+        return super().clean(column, dtype)
+
+    @property
+    def conn(self) -> Any:
+        return super().conn
+
+    def rollback(self) -> None:
+        pass
+
+    def close(self) -> None:
+        """
+        Close the underlying connection to the SQL data source if open. Else will do nothing.
+        """
+        if '_ctx' in self.__dict__:
```

Comment:
do you have any reason to override this method instead of using the implementation in the base class? https://github.com/mage-ai/mage-ai/blob/master/mage_ai/io/base.py#L315

Reply:
I was trying around some things as I was facing some errors. But looks like this was not required.
I cleaned it up now.